components/gantry/oneaxis/oneaxis.go

```diff
@@ -160,13 +164,65 @@ func newOneAxis(ctx context.Context, deps resource.Dependencies, conf resource.C
 		return nil, errors.Errorf("invalid gantry type: need 1, 2 or 0 pins per axis, have %v pins", np)
 	}
 
-	if err := oAx.home(ctx, np); err != nil {
+	if err := oAx.home(context.Background(), np); err != nil {
 		return nil, err
 	}
 
+	oAx.checkHit(context.Background())
 	return oAx, nil
 }
 
+func (g *oneAxis) checkHit(ctx context.Context) {
+	ctx, g.cancelFunc = context.WithCancel(ctx)
```

Comment:
[nit] you can create the `context.Background` in this function and just call it in the constructor without an input

Reply:
Just tried this, but `g.cancelFunc` causes an error of `expected identifier on left side of :=`. But I can break it into two lines so it would look like this. 
```
ctx := context.Background()
ctx, g.cancelFunc = context.WithCancel(ctx)
```
would you prefer this, or what's currently implemented

Reply:
what's currently implemented is fine then. You could also
```go
ctx, cancelFunc := context.WithCancel(context.Background)
g.cancelFunc = cancelFunc
```

or 
```go
var ctx context.Context // makes an empty context
ctx, g.cancelFunc := context.WithCancel(context.Background()
```

whatever looks nicest to you!

Reply:
going with the first suggestion here, thanks!